### PR TITLE
parser: support check expression in create table

### PR DESCRIPTION
**Pull request by siddontang**
nothin to do now

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -3325,6 +3326,11 @@ TableElement:
 	{
 		$$ = $1.(*coldef.TableConstraint)
 	}
+|	"CHECK" '(' Expression ')'
```
Comment (ngaut) [inline parser/parser.y]: Please add reference.

Comment (siddontang) [inline parser/parser.y]: no special reference in mysql document too, just list it in table create_definition.

So we can only refer http://dev.mysql.com/doc/refman/5.7/en/create-table.html here, but this seems no need.

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**